aries-site/src/examples/templates/dashboards/content/chartCards/RulesAudit.js

```diff
@@ -0,0 +1,191 @@
+  const [reportWindow, setReportWindow] = useState(defaultWindow);
+  const [completedAudits, setCompletedAudits] = useState(null);
+  const [nextAudit, setNextAudit] = useState(null);
+  const [rules, setRules] = useState([]);
+  const [frameworks, setFrameworks] = useState([]);
+  const [auditResults, setAuditResults] = useState([]);
+
+  // Set reporting window
+  useEffect(() => {
+    setReportWindow(prevWindow => ({
+      ...prevWindow,
+      begin: new Date(
+        new Date().setDate(
+          prevWindow.end.getDate() - REPORT_WINDOW_MAP[period],
+        ),
+      ),
+    }));
+  }, [period]);
+
+  // Isolate completed vs. upcoming audits/
+  useEffect(() => {
+    const completed = audits
+      .filter(audit => audit.completedDate)
+      .sort((a, b) => new Date(b.completedDate) - new Date(a.completedDate))
+      .filter(
+        audit =>
+          new Date(audit.completedDate) < reportWindow.end &&
+          new Date(audit.completedDate) > reportWindow.begin,
+      );
+    const next = audits
+      .sort((a, b) => new Date(a.scheduledDate) - new Date(b.scheduledDate))
+      .find(audit => new Date(audit.scheduledDate) > reportWindow.end);
+    setCompletedAudits(completed);
+    setNextAudit(next);
+  }, [audits, reportWindow.begin, reportWindow.end]);
+
+  // Get results from completed audits.
+  useEffect(() => {
+    if (completedAudits) {
+      const nextRules = checks.filter(check =>
+        completedAudits.map(audit => audit.id).includes(check.auditId),
+      );
+      const nextFrameworks = [];
+      checks.forEach(check => {
+        if (!nextFrameworks.includes(check.frameworkId)) {
+          nextFrameworks.push(check.frameworkId);
+        }
+      });
+      setRules(nextRules);
+      setFrameworks(nextFrameworks);
+    }
+  }, [checks, completedAudits]);
+
+  // Assemble chart data
+  useEffect(() => {
+    const nextAuditResults = [
+      { result: 'pass', count: 0, percent: 0, color: 'status-ok' },
+      { result: 'fail', count: 0, percent: 0, color: 'status-critical' },
+      { result: 'error', count: 0, percent: 0, color: 'status-warning' },
+    ];
+    const ruleMap = new Map();
+    nextAuditResults.forEach(result => {
+      const index = nextAuditResults.findIndex(
+        el => el.result === result.result,
+      );
+      ruleMap.set(result.result, index);
+    });
+    rules.forEach(rule => {
+      const index = ruleMap.get(rule.result);
+      nextAuditResults[index] = {
+        ...nextAuditResults[index],
+        count: nextAuditResults[index].count + 1,
+        percent: ((nextAuditResults[index].count + 1) / rules.length) * 100,
+      };
+    });
+    setAuditResults(nextAuditResults);
+  }, [rules]);
+
+  const grid = {
+    columns: ['auto', 'auto', 'auto'],
+    rows: ['auto', 'auto'],
+    areas: [
+      ['rules', 'frameworks', 'audits'],
+      ['chart', 'chart', 'chart'],
+    ],
+  };
+
+  const nextScheduledAudit = nextAudit && (
+    <Box direction="row" gap="xsmall">
```

Comment:
Below suggestion is a quick question more than a "must incorporate" -- are we intentionally aligning start in case the text wraps to 2 lines? My eye at first wanted it to be aligned center because the text is a little higher, but now i'm realizing it's probably align start for the multi-line case?


```suggestion
    <Box align="center" direction="row" gap="xsmall">
```

<img width="485" alt="Screen Shot 2022-02-23 at 8 36 34 AM" src="https://user-images.githubusercontent.com/12522275/155365564-5a9fd85d-bace-45fb-83d5-7a759ac5ba2c.png">

Reply:
That's fair -- we can leave as is.

Reply:
I went ahead and implemented `align="center"`. The text is only wrapping for a very small percentage of widths, so figure the center is better overall.

Reply:
Actually, I thought of and implemented an alternate approach. Take a look and see how you feel about it. It looks a little odd from a Grommet perspective, but translates to the DOM in a perfectly fine manner.

Reply:
I am okay with this -- wondering how we can surface the "why" (aligning line height) in documentation, maybe as part of the upcoming icon work?

Reply:
Added comment here, but it should be part of icon work as well.